blockstack/blockstackd.py

```diff
@@ -3549,8 +3601,37 @@ def run_blockstackd():
                 print "Database is NOT CONSISTENT with block {} and consensus hash {}".format(block, args.consensus_hash)
                 sys.exit(1)
 
+        print "Querying namespace IDs..."
+        namespaces_entries = db.get_all_namespace_ids()
+        namespaces_entries.sort()
+        namespaces = []
+        for namespace_str in namespaces_entries:
+            namespace_info = db.get_namespace(namespace_str)
+            namespace = {}
+            namespace['namespace_id'] = namespace_info['namespace_id']
```

Comment:
@lgalabru Is there anything else we need to export for namespaces?

Reply:
The bns contract currently needs the following fields:
https://github.com/blockstack/stacks-blockchain/blob/09badf4bb7ed570ff649ff18232650eb0c742521/src/chainstate/stacks/boot/sns.clar#L71-L83

Reply:
All variables here are now exported, except for `(renewal-rule uint)` which I can't find a corresponding value for -- what does this indicate?

Reply:
Renewal rule is a parameter that namespace buyers can pass when configuring their namespace, that will specify if names are subject to expiration (0 if not, otherwise, N for the number of blocks).

Reply:
@lgalabru I think this is the `lifetime` variable in Stacks 1.0. The namespace `lifetime` fields appear to be one of either:
`NAMESPACE_LIFE_INFINITE = 0xffffffff` or 
`EXPIRATION_PERIOD = BLOCKS_PER_YEAR`

Added in last commit.